### ISSUE TITLE
Document Translation handling creates self-relating records

### ISSUE THREAD

**Issue by maufwind** [pimcore/pimcore#4380]
# Bug Report

There should never be a `documents_translations` record which is pointing from one document to the exact same.

### Expected behavior
the column `id` and `sourceId` should not have the same value, because that relation is useless and can lead to bugs.

### Actual behavior
they can have the same value.

#### Scenario 1
Simply create a new document translation to the document itself in the backend. (yes it would be silly to do this, but why not check the id and throw an error before inserting it?)

#### Scenario 2
Let's say you have a document with the id _44_ in german and one with id _45_ in english. You link _44_ with _45_, which works well and looks like this in `documents_translations`:
```
'id', 'sourceId', 'language'
'45', '44', 'en'
```

If you open _45_, you are able to also link it. If you link _45_ with _44_ the table `documents_translations` will contain these entries:
```
'id', 'sourceId', 'language'
'44', '44', 'de'
'45', '44', 'en'
```
where the _de_ part points to itself.

Also, in general, it is not really clear why the table is build like this. Why a `language` column is needed? Why not use the document language instead? Why is the column a string instead of an id?

**Comment by mattamon** [comment]
@maufwind I agree that this is a design flaw, but does it actually have any implications? I tested it and I did not encounter any problem whatsoever.

The ids are not integers in the meanwhile.

As long as it does not affect anything I would not consider this a bug or something that we need to actively change.

**Comment by brusch** [comment]
Closing this, I agree with this [explanation](https://github.com/pimcore/pimcore/issues/4380#issuecomment-2343472700).
